Assets/LeapMotion/Scripts/Query/QueryWrapper.cs

```diff
@@ -70,29 +76,76 @@ public struct Enumerator {
     }
   }
 
+  /// <summary>
+  /// The interface all query operations must follow.  It is a modified version of the 
+  /// IEnumerator interface, optimized for speed and conciseness.
+  /// </summary>
   public interface IQueryOp<T> {
+
+    /// <summary>
+    /// Tries to get the next value in the sequence.  If this method returns true,
+    /// the next value will be placed into the out parameter t.  If this method
+    /// returns false, the sequence is at an end and t will be the default value
+    /// of T.
+    /// 
+    /// Once TryGetNext returns false, it can NEVER return true again until the
```

Comment:
This is slightly different.  In the general case, you will never interact with TryGetNext directly so you never need to worry about calling Reset().  Normally you will either use a foreach(), or a direct op like First() / ToList() / ect... all of which do not need to be manually reset.